src/Heist/Extra/Splices/Tree.hs

```diff
@@ -19,14 +19,16 @@ treeSplice ::
 treeSplice =
   go []
   where
-    go :: [a] -> (NonEmpty a -> sortKey) -> [Tree a] -> (NonEmpty a -> [Tree a] -> H.Splices (HI.Splice Identity)) -> HI.Splice Identity
+    go :: [a] -> (NonEmpty a -> [Tree a] -> sortKey) -> [Tree a] -> (NonEmpty a -> [Tree a] -> H.Splices (HI.Splice Identity)) -> HI.Splice Identity
     go pars sortKey trees childSplice = do
       let extendPars x = maybe (one x) (<> one x) $ nonEmpty pars
-      flip foldMapM (sortOn (sortKey . extendPars . rootLabel) trees) $ \(Node lbl children) -> do
+      let sorter x = sortKey (extendPars $ rootLabel x) (subForest x)
+      flip foldMapM (sortOn sorter trees) $ \(Node lbl children) -> do
```

Comment:
`let sortKey' = flip sortKey (subForest x)`? Then the existing code would require minimal changes here, and below.

Reply:
But where would the `x` come from?

Reply:
Oh, right. Okay I'll review all your PRs by this weekend.